test/throw-new-error.js

```diff
@@ -28,6 +29,11 @@ ruleTester.run('new-error', rule, {
 			errors
 		},
 		{
+			code: `throw CustomerError('foo')`,
+			output: `throw new CustomerError('foo')`,
+			errors
+		},
```

Comment:
Can you add a few tests that make sure the regex is well built? I'm thinking things like the following, but add more if you think of any

Valid
```
throw fooBarBazError('foo') // does not start with an upper case character
```

Invalid
```
throw FooBarBazError('foo') // With multiple upper case characters
throw ABCError('foo') // With multiple upper case characters in a row
throw Abc3Error('foo') // With numbers
```